### PR TITLE
Print helper state if nothing is going on

### PR DESCRIPTION
**Pull request by akoshelev**
It is an easy mistake to make to start counting records from 1 or miss sending a record. Infrastructure stays silent and does not output anything for it. This change sets up a timer for gateway loop that prints the state of helper (if there is something pending) every 10 seconds. The output looks like this

```bash
2022-12-05T05:48:59.379513Z ERROR raw_ipa::helpers::messaging: Nothing has happened on H1 in the last 10s. Here is the list of tasks pending completion:
 waiting to send: [
channel[peer=H2,step=step=protocol/run-0]: [0]
],
 waiting to receive: [
channel[peer=H2,step=step=protocol/run-0]: [1, 2, 3, 4, 5, 6, 7, 8, 9]
]
```

### CODE REVIEW

File: src/helpers/messaging.rs [hunk 1]
```diff
@@ -183,11 +190,29 @@ impl Gateway {
                                 .expect("Failed to send data to the network");
                         }
                     }
+                    _ = &mut sleep => {
+                        // TODO: split this large spawn task into smaller chunks and create a function
+                        // for it.
```
Comment (martinthomson) [inline src/helpers/messaging.rs:195]: Maybe after you get some sleep :)

**Review by martinthomson — COMMENTED** [review]
I'm going to have to try this out.  Tomorrow.

**Comment by benjaminsavage** [comment]
A few comments just on the print out format:

1. `step=step=protocol/run-0]` what is the whole `step=step=` part? Is that a duplicate?
2. There is a space in front of ` waiting to send: [`. Is that because it's printing a null in front of it or something?

**Comment by akoshelev** [comment]
> I'm getting an out of bounds error when printing the send buffer state.
I guess I should stop sending changes for review after midnight

**Comment by akoshelev** [comment]
> step=step=protocol/run-0] what is the whole step=step= part? Is that a duplicate?

yes, it is in the `ChannelId` debug implementation. I removed the extra `step=`

**Comment by akoshelev** [comment]
> There is a space in front of waiting to send: [. Is that because it's printing a null in front of it or something?

fixed that as well

**Comment by akoshelev** [comment]
> I'm getting an out of bounds error when printing the send buffer state.

Found this bug and fixed it.

```bash
waiting to send: [
channel[peer=H2,step=protocol/run-0]: [5, 6]
],
waiting to receive: [
channel[peer=H2,step=protocol/run-0]: [1, 2, 3, 4, 5, 6, 7, 8, 9]
]
```

**Comment by martinthomson** [comment]
Did you miss https://github.com/akoshelev/raw-ipa/pull/1 then?

**Comment by akoshelev** [comment]
oh yea, I did. For some reason I am not getting notified for PRs sent to my fork, lol

**Comment by akoshelev** [comment]
All right I  think this one is good for review

```bash
2022-12-07T06:00:46.514391Z ERROR raw_ipa::helpers::messaging: List of tasks pending completion on H1:
waiting to send: [
    channel[H2,step=protocol/run-0]: [5, 6]
],
waiting to receive: [
    channel[H2,step=protocol/run-0]: [1, 2, 3, 4, 5, 6, 7, 8, 9]
]
```

**Pull request merged**